pkg/azuredx/azureauth/auth_test.go

```diff
@@ -86,6 +86,32 @@ func TestOnBehalfOf(t *testing.T) {
 	}
 }
 
+func TestOnBehalfOfDisabled(t *testing.T) {
+	// compose request
+	var req backend.QueryDataRequest
+	req.PluginContext.User = &backend.User{Login: "alice"}
+	req.Headers = make(map[string]string)
+	req.Headers["X-ID-Token"] = "ID-TOKEN"
+
+	// setup & test
+	fakeTokenProvider := &FakeTokenProvider{}
+
+	c := &ServiceCredentialsImpl{
+		tokenProvider:  fakeTokenProvider,
+	}
+	c.OnBehalfOf = false
+
+	auth, err := c.QueryDataAuthorization(context.Background(), &req)
+
+	if err != nil {
+		t.Errorf("got error %q", err)
+	}
```

Comment:
another, shorter, way of writing this: `require.NoError(t, err)`

`require` from `"github.com/stretchr/testify/require"`